src/rust/engine/process_execution/src/remote/streaming.rs

```diff
@@ -0,0 +1,590 @@
+          wait_execution_request.set_name(operation_name.to_owned());
+          self
+            .execution_client
+            .wait_execution_opt(&wait_execution_request, call_opt)
+        }
+      };
+
+      // Take action based on whether we received an output stream or whether there is an
+      // error to resolve.
+      let actionable_result = match rpc_result {
+        Ok(operation_stream) => {
+          // Monitor the operation stream until there is an actionable operation
+          // or status to interpret.
+          let compat_stream = operation_stream.compat();
+          let (operation_name_opt, actionable_result) =
+            self.wait_on_operation_stream(compat_stream).await;
+
+          // Save the operation name in case we need to reconnect via WaitExecution.
+          if let Some(name) = operation_name_opt {
+            current_operation_name = Some(name);
+          }
+
+          actionable_result
+        }
+        Err(err) => match err {
+          grpcio::Error::RpcFailure(rpc_status) => {
+            let mut status = Status::new();
+            status.code = rpc_status.status.into();
+            OperationOrStatus::Status(status)
+          }
+          _ => {
+            return Err(format!("gRPC error: {}", err));
+          }
+        },
+      };
+
+      match self.extract_execute_response(actionable_result).await {
+        Ok(result) => return Ok(result),
+        Err(err) => match err {
+          ExecutionError::Fatal(e) => return Err(e),
+          ExecutionError::Retryable(e) => {
+            // do nothing, will retry
+            trace!("retryable error: {}", e);
+          }
+          ExecutionError::MissingDigests(missing_digests) => {
+            trace!(
+              "Server reported missing digests; trying to upload: {:?}",
+              missing_digests,
+            );
+
+            let _ = self
+              .store
+              .ensure_remote_has_recursive(missing_digests)
+              .compat()
+              .await?;
+          }
+          ExecutionError::NotFinished(_) => unreachable!(),
+        },
+      }
+    }
+  }
+}
+
+#[async_trait]
+impl crate::CommandRunner for StreamingCommandRunner {
+  /// Run the given MultiPlatformProcess via the Remote Execution API.
+  async fn run(
+    &self,
+    request: MultiPlatformProcess,
+    context: Context,
+  ) -> Result<FallibleProcessResultWithPlatform, String> {
+    // Construct the REv2 ExecuteRequest and related data for this execution request.
+    let compatible_underlying_request = self.extract_compatible_request(&request).unwrap();
+    let store = self.store.clone();
+    let (action, command, execute_request) =
+      super::make_execute_request(&compatible_underlying_request, self.metadata.clone())?;
+    let Process {
+      description,
+      timeout,
+      input_files,
+      ..
+    } = compatible_underlying_request;
+
+    debug!("Remote execution: {}", description);
```

Comment:
At `debug`, dumping the whole request wouldn't hurt even.

Reply:
I may move this back to trace. I set some of these to debug so I could avoid log spam from trace logging of everything that is not remote execution related. (Being able to increase the log level of subsets of the log statements would be useful. Some sort of logger name concept would do.)